spec/functional/resource/windows_task_spec.rb

```diff
@@ -0,0 +1,458 @@
+      end
+    end
+
+    context "when a non-system user is passed without password" do
+      it "raises error" do
+        subject.user "Administrator"
+        subject.frequency :onstart
+        expect { subject.after_created }.to raise_error("Can't specify a non-system user without a password!")
+      end
+    end
+
+    context "when interactive_enabled is passed for a System user without password" do
+      it "raises error" do
+        subject.interactive_enabled true
+        subject.frequency :onstart
+        expect { subject.after_created }.to raise_error("Please provide the password when attempting to set interactive/non-interactive.")
+      end
+    end
+
+    context "when frequency_modifier > 1439 is passed for frequency=:minute" do
+      it "raises error" do
+        subject.frequency_modifier 1450
+        subject.frequency :minute
+        expect { subject.after_created }.to raise_error("frequency_modifier value 1450 is invalid.  Valid values for :minute frequency are 1 - 1439.")
+      end
+    end
+
+    context "when invalid months are passed" do
+      it "raises error" do
+        subject.months "xyz"
+        subject.frequency :monthly
+        expect { subject.after_created }.to raise_error("months attribute invalid. Only valid values are: JAN, FEB, MAR, APR, MAY, JUN, JUL, AUG, SEP, OCT, NOV, DEC and *. Multiple values must be separated by a comma.")
+      end
+    end
+
+    context "when idle_time > 999 is passed" do
+      it "raises error" do
+        subject.idle_time 1000
+        subject.frequency :on_idle
+        expect { subject.after_created }.to raise_error("idle_time value 1000 is invalid.  Valid values for :on_idle frequency are 1 - 999.")
+      end
+    end
+
+    context "when idle_time is passed for frequency=:monthly" do
+      it "raises error" do
+        subject.idle_time 300
+        subject.frequency :monthly
+        expect { subject.after_created }.to raise_error("idle_time attribute is only valid for tasks that run on_idle")
+      end
+    end
+  end
+
+  describe "action :delete" do
+    subject do
+      new_resource = Chef::Resource::WindowsTask.new(task_name, run_context)
+      new_resource.command task_name
+      new_resource
+    end
+
+    it "deletes the task if it exists" do
+      subject.run_action(:create)
+      delete_task
+      task_details = windows_task_provider.send(:load_task_hash, task_name)
+      expect(task_details).to eq(false)
+    end
+  end
+
+  describe "action :run" do
+    after { delete_task }
+
+    subject do
+      new_resource = Chef::Resource::WindowsTask.new(task_name, run_context)
+      new_resource.command task_name
+      new_resource
+    end
+
+    it "runs the existing task" do
+      subject.run_action(:create)
+      subject.run_action(:run)
+      task_details = windows_task_provider.send(:load_task_hash, task_name)
+      expect(task_details[:Status]).to eq("Ready")
```

Comment:
@mwrock , @btm, I previously had `expect(task_details[:Status]).to eq("Running")` condition here, but Appveyor was returning the Status as `Ready` for the task even after running `subject.run_action(:run)`.

Please suggest how can we debug this issue for Appveyor system.